docs/installation/integrations/next-js.md

```diff
@@ -0,0 +1,204 @@
+
+```bash
+yarn add @ckeditor/ckeditor5-react
+```
+
+Next, we will use the installed dependencies in a React component. Create a new component in the components directory, for example, `components/custom-editor.js`. Inside the component file, import all necessary dependencies. Then, create a functional component that returns the CKEditor&nbsp;5 React component with a proper configuration.
+
+```jsx
+// components/custom-editor.js
+
+import React from 'react';
+import { CKEditor } from "@ckeditor/ckeditor5-react";
+import Editor from "ckeditor5-custom-build";
+
+const editorConfiguration = {
+    toolbar: [
+        'heading',
+        '|',
+        'bold',
+        'italic',
+        'link',
+        'bulletedList',
+        'numberedList',
+        '|',
+        'outdent',
+        'indent',
+        '|',
+        'imageUpload',
+        'blockQuote',
+        'insertTable',
+        'mediaEmbed',
+        'undo',
+        'redo'
+    ]
+};
+
+function CustomEditor( props ) {
+		return (
+			<CKEditor
+				editor={ Editor }
+				config={ editorConfiguration }
+				data={ props.initialData }
+				onChange={ (event, editor ) => {
+					const data = editor.getData();
+					console.log( { event, editor, data } );
+				} }
+			/>
+		)
+}
+
+export default CustomEditor;
+```
+
+Our `CustomEditor` component is ready to be used inside a page. The usage will differ depending on the chosen routing strategy.
+
+### Page Router
+
+You can create a second page inside the `pages` directory or replace the content of the existing `index.js` page &ndash; we will go with the second option. CKEditor&nbsp;5 is a client-side text editor and relies on the browser APIs, so we need to disable SSR for our custom component. We can lazily load the component using the `dynamic()` function built into Next.js.
+
+```jsx
+// pages/index.js
+
+import React from 'react';
+import dynamic from 'next/dynamic';
+
+const CustomEditor = dynamic( () => {
+  return import( '../components/custom-editor' );
+}, { ssr: false } );
+
+function Home() {
+  return (
+    <CustomEditor 
+      initialData='<h1>Hello from CKEditor in Next.js!</h1>'
+      onChange={ ( e ) => console.log( e ) }
+    />
+  );
+}
+
+export default Home;
+```
+
+### App Router
+
+App Router, by default, uses server components. It means we need to mark a component as client-side explicitly. We can achieve that by using the `'use client'` directive at the top of a file, above your imports. Also, pay attention to the folder structure, which is different now.
+
+```jsx
+// app/page.js
+
+'use client'
+
+import React from 'react';
+import dynamic from 'next/dynamic';
+
+const CustomEditor = dynamic( () => {
+  return import( '../components/custom-editor' );
+}, { ssr: false } );
+
+function Home() {
+  return (
+    <CustomEditor 
+      initialData='<h1>Hello from CKEditor in Next.js!</h1>'
+      onChange={ ( e ) => console.log( e ) }
+    />
+  );
+}
+
+export default Home;
+```
+
```

Comment:
App router and pages are so similar that I wonder if we shouldn't have one code block like:

```javascript
'use client' // only in App router.

import React from 'react';
import dynamic from 'next/dynamic';

const CustomEditor = dynamic( () => {
  return import( '../components/custom-editor' );
}, { ssr: false } );
// . . .
```

And join both sections together for brevity. Up to you, I see pros and cons for both approaches.

Reply:
Eventually, I decided to merge these snippets. The difference in the folder structure is visible by showing the final state for both routers. I think it adds more value than the initial state. WDYT?